File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
    ListNode t=headA;
    HashSet<ListNode>a=new HashSet<>();
    while(t!=null)
    {a.add(t);
    t=t.next;
    }
    t=headB;
    while(t!=null)
    {if(a.contains(t)) return t;
    t=t.next;
    }   
  return null;
    }
}

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public: 
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
    std::unordered_set<ListNode*>a;
    ListNode* t=headA;
    while(t!=NULL)
    {a.insert(t);
    t=t->next;
    }
    t=headB;
    while(t!=NULL)
    {if(a.count(t)) return t;
    t=t->next;
    }
    return NULL;    
    }
};